### ISSUE TITLE
Improve ability to access ParentList item associated with a row

### ISSUE THREAD

**Issue by fichna** [bignerdranch/expandable-recycler-view#157]
Note from @paul-turner:
Editing this issue to track future work.

ParentViewHolder could have helper methods that keep a reference to the parent and child position and the item that is currently associated with it. ChildViewHolder could benefit from the same information.

**Original Title:** Can I get index of clicked parent? 
**Original Text:**

Hi, I'm using this library and it's great. 
I want add image button to parent view which opens popup window for some additional actions. 
But I have no idea how to get parent item that is clicked. I can get adapterPosition on viewholder and send it through listener to adapter. But adapterPosition is absolute position in adapter... to get my object I must count only parent views, not parent and child.

Is there some method to get position of only parent views? 
Or method which returns me list of all displayed items?
Please help me

**Comment by fichna** [comment]
I tried use `getListItem(adapterPosition)` but it returns ParrentWrapper and I don't know how to get my parent object from it...

**Comment by fichna** [comment]
I found a solution probably... but it is realy ugly...
`((Recipe) ((ParentWrapper) getListItem(adapterPosition)).getParentListItem()).getId()`

**Comment by paul-turner** [comment]
Thanks @fichna, in `onBindParentViewHolder` we give the `ParentListItem` associated with that view, you can store that as a variable in your ParentViewHolder so that it's not as ugly a chain of methods.

That's the current best solution I've come up with right now, let me know if it doesn't fit your needs.

**Comment by mrkidjordje** [comment]
i have similar problem, i want to notify child data changed, but i can't get parent and child locations? how can I get this data so i could pass it to method?

**Comment by fichna** [comment]
It would be nice to have some example of this

**Comment by mrkidjordje** [comment]
Hi, I'm relativity new to Android programming, so i don't know what days you need, in my child view i have button and when i press it,  some communication with server is done and if response is successful, i want to change color and text on that button in child view. For that there is a method from this library, but i need to pass parent and child position as parameters, so i could notify data set changed on that view. (In this case text and background color)

**Comment by mrkidjordje** [comment]
It's not days, is should be data, autocorrect made mistake.

**Comment by paul-turner** [comment]
Thanks for the feedback @fichna and @mrkidjordje, I will see about improving our documentation around this area (possibly improving the library itself to make this easier to configure).

I'm going to rename this issue and use it for tracking said work.

**Comment by paul-turner** [comment]
With #171 there is now a snapshot build that should let you access more information from the `ViewHolder`s:
- Added ParentViewHolder methods:
    - `getParentListItem()`
    - `getParentAdapterPosition()`
- Added ChildViewHolder methods:
    - `getChildListItem()`
    - `getParentAdapterPosition()`
    - `getChildAdapterPosition()`

This required changing the package location so bumped up the version to 3.0.0, which means this will probably be a snapshot as we get more features in for the library.
Instructions on how to get the snapshot are available in the `README` at https://github.com/bignerdranch/expandable-recycler-view/tree/dev

Hope this helps everyone with their logic.

**Comment by schelbin** [comment]
Hello. I am trying to use the:

    compile 'com.bignerdranch.android:expandablerecyclerview:3.0.0-SNAPSHOT'

to use the new methods and get my currect parent possition, but when builded, I get this error:

Error:(31, 13) Failed to resolve: com.bignerdranch.android:expandablerecyclerview:3.0.0-SNAPSHOT

There are any error in the syntaxis of the version that I ang trying to get?

Thanks

**Comment by schelbin** [comment]
Hello... Any idea why the compile 'com.bignerdranch.android:expandablerecyclerview:3.0.0-SNAPSHOT' don't funtion and 2.1.1 yes? thanks

**Comment by paul-turner** [comment]
@schelbin Hi there,

Did you add the snapshot repository? Instructions are the following:

Add Sonatype's snapshots to your repositories closure in the root `build.gradle`:
 ```gradle
 allprojects {
     repositories {
         maven { url "https://oss.sonatype.org/content/repositories/snapshots" }
     }
 }
 ```

**Comment by paul-turner** [comment]
Hello, we've released 3.0.0-RC1 with a fix for this issue. 

I would love your feedback with any issues you have with the solution before we make 3.0.0 final. Check out the Pull Request that contains the fix for your issue (should be referenced above), and the [changelog](https://github.com/bignerdranch/expandable-recycler-view/releases/tag/v3.0.0-RC1) for a list of changes that have come with 3.0.0 (there are a lot of them!). 

If you have any issues feel free to re-open this issue with details of what you are seeing.

As a note, the method names have changes since the release above.
